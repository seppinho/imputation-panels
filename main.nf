#!/usr/bin/env nextflow
/*
========================================================================================
    genepi/nf-imputationserver-refpanels
========================================================================================
    Github : https://github.com/genepi/nf-imputationserver-refpanels
    Author: Sebastian Sch√∂nherr / Lukas Forer
    ---------------------------
*/

nextflow.enable.dsl = 2


include { IMPUTATION_PANELS } from './workflows/imputation_panels'

workflow {
    IMPUTATION_PANELS ()
}

