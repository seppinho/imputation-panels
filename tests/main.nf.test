@Grab(group='net.lingala.zip4j', module='zip4j', version='2.11.5')
import net.lingala.zip4j.ZipFile;

nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Should create three msav files") {

        when {
            params {
                project  = "test-job"
                files = "$projectDir/tests/input/three/*.vcf.gz"
            }   
        }

        then {
            assert snapshot(workflow.trace).match()
            assert workflow.success
        }

    }



}