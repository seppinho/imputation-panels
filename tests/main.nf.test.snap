{
    "Should create three msav files": {
        "content": [
            {
                "tasksFailed": 0,
                "tasksCount": 3,
                "tasksSucceeded": 3
            }
        ],
        "timestamp": "2023-11-10T17:02:16.158553"
    }
}